### DISCUSSION TITLE
Just found the PS4 kernel page size is 4K, not 16K

### DISCUSSION THREAD

**Discussion by ultimaweapon** [obhq/obliteration#958] [General]
![Screenshot from 2024-09-01 16-32-15](https://github.com/user-attachments/assets/3ca83a2f-89d3-467a-9e7d-e668e738beb5)

The corresponding call in FreeBSD:

```c
        wakeaddr = contigmalloc(4 * PAGE_SIZE, M_DEVBUF, M_WAITOK, 0x500,
            0xa0000, PAGE_SIZE, 0ul);
```

I wonder why the PS4 emulate 16K page size in the other places.

**Comment by red-prig** [comment]
Porting problem? Does ps4 even have acpi? Will this code even run?

**Reply by ultimaweapon** [reply]
AFAICT the PS4 has ACPI. Recently I found out the x86-64 does not support 16K page size at a hardware level while developing a hypervisor to run our kernel mode kernel. This issue bugged me for awhile because everywhere I saw in the PS4 kernel use 16K page size. My first theory is Sony request this special page size from AMD but it still does not make sense since adding support for 16K page size on x86-64 requires too much changes to the CPU.

**Reply by red-prig** [reply]
Well, 16 KB can work only at the user mode level, why they did it this way is not very clear
